src/components/addition/DetailedCard.tsx

```diff
@@ -0,0 +1,51 @@
+import { ProductData, ProductsResponse } from '../../interfaces';
+import { useRouter } from 'next/router';
+
+function DetailedCard({ data }: ProductsResponse) {
+  const router = useRouter();
+  const { page, limit, search, details } = router.query;
+  const targetProductObj = data?.results?.find(
+    (product: ProductData) => details === product.name.en
+  );
+  const images = targetProductObj?.masterVariant.images;
+  return (
+    <div
+      data-testid="detailed-container"
+      className="detailed-product-container"
+    >
+      <div className="product-page">
+        <button
```

Comment:
Maybe split for some components and separate functions for handlers? Your code is overloaded...